### ISSUE TITLE
IE9 EXT 4.1.1 / BoxSelect 2.0.1 Modal Window

### ISSUE THREAD

**Issue by ValterBorges** [kveeiv/extjs-boxselect#13]
BoxSelect 2.0.1 and Ext 4.1.1 on a form inside a popup window causing window to not render correctly.

This worked with EXT 4.0.7 and BoxSelect 1.3.1.

The Window def:

Ext.create('Ext.window.Window',
    {
        id: 'MyEditForm_Win',
        closeAction: 'hide',
        modal: false,
        frame: false,
        resizable: false,
        width: 860,
        bodyPadding: 5,
        title: 'My Detail',
        bodyStyle: { "background-color": "#FFFFFF" },
        items: [
            { height: 10, border: false },
            Ext.create('Ext.form.Panel',
                {
                    itemId: 'MyEdit_Form', 
                    width: 840,
                    trackResetOnLoad: true,
                    defaultType: 'panel',
                    frame: false,
                    border: true,
                    title: '&nbsp;',
                    style: 'background:#FFFFFF;',
                    bodyStyle: { "background-color": "#FFFFFF" },
                    items: [
                            Ext.create("My.view.fieldset.MyEdit_Fieldset",
                                { 
                                    itemId: 'MyEdit_Fieldset',
                                    title: null, 
                                    border: false
                                }
                             )
                            ]
                    , tbar: [
                            { itemId: 'btnMyDetail_History', hidden: false, icon: cdnUrl + '/cdn/images/My/icons/list.gif', tooltip: 'History', text: 'History' }
                            '->'
                         ]
            }),
           { height: 10, border: false }
           ]
    });

The fieldset with BoxSelect which is used in the window.

Ext.define("My.view.fieldset.MyEdit_Fieldset", {
    extend: "Ext.panel.Panel",
    initComponent: function () {
        Ext.apply(this, this.initialConfig,
         {
             title: '',
             collapsible: false,
             border: false,
             //cls: 'lrgPanelHeader',
             layout: {
                 type: 'table',
                 columns: 2,
                 tableAttrs: {
                     style: {
                         width: '100%',
                         margin: '0px',
                         padding: '0px',
                         border: '0px'
                     }
                 },
                 tdAttrs: {
                     style: {
                         padding: '2px 2px 1px 4px',
                         width: '50%',
                         verticalAlign: 'top'
                     }
                 },
                 cellCls: 'x-panel-body-default'
             },
             defaults: {
                 margins: '0 0 0 0',
                 padding: '0 0 0 0',
                 style: 'border: solid 0px black;',
                 labelStyle: 'font-weight:bold;font-size:12px;',
                 labelAlign: 'left'
             },
             items: [
               {
                   border: false,
                   layout: {
                       type: 'table',
                       columns: 2,
                       tableAttrs: {
                           style: {
                               width: '100%',
                               margin: '0px',
                               padding: '0px',
                               border: '1px'
                           }
                       },
                       tdAttrs: {
                           style: {
                               padding: '0px 10px 0px 0px',
                               width: '50%',
                               verticalAlign: 'top',
                               border: '2px'
                           }
                       }
                   },
                   defaults: {
                       labelStyle: 'font-weight:bold;font-size:12px;'
                   },
                   items: [
                        Ext.create("Ext.form.field.Display", {
                            itemId: 'Age',
                            name: 'Age',
                            fieldLabel: 'Age',
                            labelWidth: 30,
                            labelAlign: 'left',
                            width: 200
                        }),
                        Ext.create("Ext.form.field.Display", {
                            itemId: 'DOB',
                            name: 'Dob',
                            fieldLabel: 'DOB',
                            labelWidth: 30,
                            labelAlign: 'left',
                            width: 200
                        })
                        ]
               },
               {
                   border: false,
                   layout: {
                       type: 'table',
                       columns: 2,
                       tableAttrs: {
                           style: {
                               width: '100%',
                               margin: '0px',
                               padding: '0px',
                               border: '0px'
                           }
                       },
                       tdAttrs: {
                           style: {
                               padding: '0px 4px 0px 0px',
                               width: '50%'
                           }
                       }
                   },
                   defaults: {
                       labelStyle: 'font-weight:bold;font-size:12px;'
                   },
                   items: [
                Ext.create("Ext.form.field.Display", {
                    itemId: 'Gender',
                    name: 'CD_GNDR_DESC_M',
                    fieldLabel: 'Gender',
                    labelWidth: 50,
                    width: 125,
                    labelAlign: 'left'
                }),
                Ext.create("Ext.form.field.Display", {
                    itemId: 'Primary_Language',
                    name: 'CD_LNG_PRFR_DESC_L',
                    fieldLabel: 'Primary Language',
                    labelWidth: 115,
                    width: 325,
                    labelAlign: 'left'
                })]
               },
               Ext.create("Ext.form.field.Display", {
                   itemId: 'Race',
                   name: 'Race',
                   fieldLabel: 'Race',
                   labelWidth: 35,
                   width: 325,
                   labelAlign: 'left'
               }),
               Ext.create("Ext.form.field.Display", {
                   itemId: 'Ethnicity',
                   name: 'Ethnicity',
                   fieldLabel: 'Ethnicity',
                   labelWidth: 60,
                   width: 325,
                   labelAlign: 'left'
               }),
               Ext.create('Ext.ux.form.field.BoxSelect', {
                   itemId: 'Educationlevel_Combo',
                   name: 'Education_Level',
                   forceSelection: true,
                   autoSelect: false,
                   typeAhead: true,
                   queryMode: 'local',
                   displayField: 'l_desc',
                   valueField: 'ID_GRPI',
                   fieldLabel: 'Education/Development Level',
                   labelAlign: 'top',
                   labelWidth: 100,
                   width: 375,
                   autoHeight: true,
                   growMin: 50,
                   growMax: 150,
                   validator: function (value) {

```
                   //called first before other validation
                   //on blur causes validation therefore don't change focus because it will cause recursion
                   //accessing .getValueRecords causes item close issues.

                   //this.clearInvalid();

                   if (Ext.isEmpty(value)) {

                       return true;
                   }

                   /*
                   .   If Daycare, Birth to Three, Head Start, Pre-K or Kindergarten are selected then multiple selections can be made.  
                   All others (1st grade and above), only one selection can be made.
                   */



                   //var selectedArray = this.getValue();
                   var selectedRecords = this.getValueRecords();
                   //var selectedArray = value.split(",");

                   //check multiple selections and nothing else other than Daycare, Birth to Three, Head Start, Pre-K or Kindergarten
                   //if (!Ext.isEmpty(selectedArray) && selectedArray.length > 1) {
                   if (!Ext.isEmpty(selectedRecords) && selectedRecords.length > 1) {
                       //var store = this.store;

                       var i = 0;
                       var record;

                       //Get selectedRecords
                       //selectedRecords = this.getValueRecords();

                       var countMultipleGroup = 0;
                       var countSingleGroup = 0;
                       for (i = 0; i < selectedRecords.length; i++) {
                           var sdesc = selectedRecords[i].get("s_desc");

                           if (sdesc === "HSTR" || sdesc === "PREK" || sdesc === "KIND" || sdesc === "DAYC" || sdesc === "BRTH") {

                               countMultipleGroup++;

                           }
                           else {
                               countSingleGroup++;
                           }

                           if (countSingleGroup > 1 || (countSingleGroup > 0 && countMultipleGroup > 0)) {
                               return "Multiple selections allowed only for combinations of the following:<br/>(Daycare, Birth to Three, Head Start, Pre-K or Kindergarten)";
                           }

                       }

                   }

                   return true;
               }
           }),
           {
               border: false,
               defaults: {
                   labelStyle: 'font-weight:bold;font-size:12px;'
               },
               layout: 'fit',
               items: [

                { border: false, height: 10, width: 1 },
            Ext.create("Ext.form.field.Text", {
                itemId: 'City',
                name: 'TX_TOWN',
                fieldLabel: 'City',
                labelWidth: 50,
                width: 325,
                labelAlign: 'left',
                maxLength: 40,
                enforceMaxLength: true
            }),
                {
                    height: 10,
                    border: false
                },
            Ext.create('Ext.form.ComboBox', {
                itemId: 'State_Combo',
                name: 'CD_STATE',
                //forceSelection: true,
                //autoSelect: true,
                //typeAhead: false,
                queryMode: 'local',
                //enableKeyEvents: true,
                displayField: 's_desc_l_desc',
                valueField: 'CD_STATE',
                fieldLabel: 'State',
                labelAlign: 'left',
                labelWidth: 50,
                width: 325
            }),

            {
                height: 10,
                border: false
            },
            Ext.create("Ext.form.field.Date", {
                itemId: 'Most_Recent_Removal_Date',
                name: 'DT_LST_RMVL',
                fieldLabel: 'Most Recent Removal Date',
                labelWidth: 175,
                labelAlign: 'left',
                width: 300
            }),
                {
                    height: 10,
                    border: false
                },
                {
                    layout: 'fit',
                    border: false,
                    height: 60,
                    items: [
            {
                xtype: 'label',
                itemId: 'lblTimeInCare',
                text: 'Time in Care in the last 22 consecutive months',
                style: 'font-weight:bold;font-size:12px;'
            },
            {
                layout: 'hbox',
                border: false,
                height: 30,
                items: [
                Ext.create("Ext.form.field.Number", {
                    itemId: 'Time_In_Care_Months',
                    name: 'QT_TIME_IN_CARE_MM',
                    fieldLabel: 'months(s)',
                    labelWidth: 75,
                    labelAlign: 'right',
                    width: 150,
                    minValue: 0,
                    maxValue: 22
                }), {
                    width: 20,
                    height: 30,
                    border: false
                },
                Ext.create("Ext.form.field.Number", {
                    itemId: 'Time_In_Care_Days',
                    name: 'QT_TIME_IN_CARE_DD',
                    fieldLabel: 'days(s)',
                    labelWidth: 75,
                    labelAlign: 'right',
                    width: 150,
                    minValue: 0,
                    maxValue: 31
                })]
            }
            ]
                }
        ]
           },
            Ext.create('Ext.form.ComboBox', {
                itemId: 'Placement_Setting_Combo',
                name: 'CD_PLCM_SETNG',
                forceSelection: true,
                autoSelect: false,
                typeAhead: true,
                queryMode: 'local',
                displayField: 'l_desc',
                valueField: 'ID_GRPI',
                labelAlign: 'top',
                pickerAlign: "tl-bl",
                fieldLabel: 'Placement Setting',
                labelWidth: 125,
                width: 300
            }),
            Ext.create('Ext.form.ComboBox', {
                itemId: 'Legal_Status_Combo',
                name: 'CD_LGL_STAT',
                forceSelection: true,
                autoSelect: false,
                typeAhead: true,
                queryMode: 'local',
                displayField: 'l_desc',
                valueField: 'ID_GRPI',
                labelAlign: 'top',
                fieldLabel: 'Legal Status',
                labelWidth: 100,
                width: 300
            }),
            Ext.create('Ext.form.ComboBox', {
                itemId: 'Permanency_Goal_Combo',
                name: 'CD_PRMNCY_GOAL',
                forceSelection: true,
                autoSelect: false,
                typeAhead: true,
                queryMode: 'local',
                displayField: 'l_desc',
                valueField: 'ID_GRPI',
                fieldLabel: 'Permanency Goal',
                labelAlign: 'top',
                labelWidth: 125,
                width: 350
            }),
            Ext.create('Ext.form.ComboBox', {
                itemId: 'Concurrent_Goal_Combo',
                name: 'CD_CNCRNCY_GOAL',
                forceSelection: true,
                autoSelect: false,
                typeAhead: true,
                queryMode: 'local',
                displayField: 'l_desc',
                valueField: 'ID_GRPI',
                fieldLabel: 'Concurrent Goal',
                labelAlign: 'top',
                labelWidth: 105,
                width: 350
            }),
            Ext.create("Ext.form.RadioGroup", {
                itemId: 'Previously_Adopted',
                fieldLabel: 'Previously Adopted',
                labelWidth: 125,
                labelAlign: 'left',
                layout: 'hbox',
                style: {
                    verticalAlign: 'top'
                },
                defaults: {
                    margins: '0 0 0 10',
                    padding: '0 0 0 0'
                },
                width: 250,
                items: [
            {
                boxLabel: 'Yes',
                name: 'CD_PREV_ADPT',
                inputValue: 1
            }, {
                boxLabel: 'No',
                name: 'CD_PREV_ADPT',
                inputValue: 2
            }]
            }),
            Ext.create("Ext.form.RadioGroup", {
                itemId: 'International_Adoption',
                fieldLabel: 'International Adoption',
                labelWidth: 150,
                labelAlign: 'left',
                layout: 'hbox',
                style: {
                    verticalAlign: 'top'
                },
                defaults: {
                    margins: '0 0 0 10',
                    padding: '0 0 0 0'
                },
                width: 250,
                items: [
            {
                boxLabel: 'Yes',
                name: 'CD_FRGN_ADPT',
                inputValue: 1
            }, {
                boxLabel: 'No',
                name: 'CD_FRGN_ADPT',
                inputValue: 2
            }]
            }),
            Ext.create("Ext.form.RadioGroup", {
                itemId: 'Bio_Father_Identity_Unknown',
                fieldLabel: 'Biological Father\'s Identity is Unknown',
                labelWidth: 250,
                labelAlign: 'left',
                layout: 'hbox',
                style: {
                    verticalAlign: 'top'
                },
                defaults: {
                    margins: '0 0 0 10',
                    padding: '0 0 0 0'
                },
                width: 350,
                items: [
            {
                boxLabel: 'Yes',
                name: 'CD_FTHR_UNKNOWN',
                inputValue: 1
            }, {
                boxLabel: 'No',
                name: 'CD_FTHR_UNKNOWN',
                inputValue: 2
            }]
            }),
            Ext.create("Ext.form.RadioGroup", {
                itemId: 'Bio_Mother_Identity_Unknown',
                fieldLabel: 'Biological Mother\'s Identity is Unknown',
                labelWidth: 250,
                labelAlign: 'left',
                layout: 'hbox',
                style: {
                    verticalAlign: 'top'
                },
                defaults: {
                    margins: '0 0 0 10',
                    padding: '0 0 0 0'
                },
                width: 350,
                items: [
            {
                boxLabel: 'Yes',
                name: 'CD_MTHR_UNKNOWN',
                inputValue: 1
            }, {
                boxLabel: 'No',
                name: 'CD_MTHR_UNKNOWN',
                inputValue: 2
            }]
            }),
            {
                height: 50,
                border: false,
                margin: 10,
                colspan: 2,
                width: 800,
                items: [
            Ext.create('Ext.Button', {
                itemId: 'btnEditReset',
                text: 'Reset',
                scale: 'large',
                icon: cdnUrl + '/cdn/images/my/icons/small-reset.png',
                style: 'float:left;margin-left:10px;',
                border: false
            }), Ext.create('Ext.Button', {
                itemId: 'btnEditSave',
                text: 'Save',
                scale: 'large',
                icon: cdnUrl + '/cdn/images/my/save1g.jpg',
                style: 'float:right;margin-right:10px;margin-top:10px;margin-bottom:10px',
                border: false
            })]
            }
        ]
     });
    this.callParent(arguments);
}
```

});

**Comment by ValterBorges** [comment]
After some further review this is actually running in IE9 under IE8 compatibility mode.  So it's likely the same issue others have already reported with IE7/IE8 not rendering correctly.

**Comment by kveeiv** [comment]
2.0.1 is the latest and works against 4.1.0 (untested against 4.1.1), and 2.0.2 is coming shortly (tested against 4.1.0 and 4.1.1).

Can you verify that your document has the appropriate <!DOCTYPE code at the top? I use the included examples/boxselect.html as a test (which is also the main example linked to from the repository home page), and have no issues with IE8.

**Comment by ValterBorges** [comment]
Been using extjs since 1.0, 2.x, 3.x, 4.x now about 6 years.  Here is what our document type has always looked like.
 so it works with IE7 - IE9, FF 3 - 12, Chrome, Safari.

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
<head>
    <meta http-equiv="X-UA-Compatible" content="IE=8" />

**Comment by ValterBorges** [comment]
When I change boxselect.html to use the doctype above it works in FF but IE it blows up on line 
37147 of ext-all-debug.js.

Tommorrow I will go thru some other doctypes and report back.
http://hsivonen.iki.fi/doctype/

Thanks,
Valter

**Comment by ValterBorges** [comment]
The error on line 37147 was a document origin issue when trying to connect in IE on the example i believe it's unrelated.

I changed the example to use the following doctype and meta info and the example works fine.

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
<head>
    <meta http-equiv="X-UA-Compatible" content="IE=8" />

I more convinced now that it has something to do with forms, form panels and modal windows.

Can you try it with a stripped down version of my example window and config.

Thanks,
Valter

**Comment by kveeiv** [comment]
Can you create a stripped down version and post it to https://gist.github.com/ ?

**Comment by kveeiv** [comment]
Can you try the following change? This resolved another IE7/8 issue and might resolve yours as well:

I put out BoxSelect 2.0.2 yesterday, if you can change line 1627:

```
owner.inputElCt.setStyle('width', inputWidth + 'px');
```

To this:

```
if (inputWidth > 0) {
  owner.inputElCt.setStyle('width', inputWidth + 'px');
}
```

and let me know the results, this might help solve your issue (though having not seen the issue yet, I'm not 100% sure on this).

**Comment by kveeiv** [comment]
Could you help me test another possible solution? There was a report on the forum threads about another width issue, and I am curious as to whether or not it will also resolve your IE issue. As I don't have a test case to replicate this, it is difficult for me to test. The change to test would be to replace the entire layout class at the bottom of BoxSelect.js with the layout class given in this post: http://www.sencha.com/forum/showthread.php?134751-Ext.ux.form.field.BoxSelect-Intuitive-Multi-Select-ComboBox&p=853134&viewfull=1#post853134

I would be very appreciative if you could test this with your case, instead of just the "if" check that I had you add, as it is probably a more comprehensive solution if it works.

**Comment by ValterBorges** [comment]
Great, ill check it out tommorrow.

**Comment by ValterBorges** [comment]
kveeiv,
I used boxselect 2.0.2 with EXTJS 4.1.1 and modified the layout according to the above thread and it fixes the issues I was having.

**Comment by kveeiv** [comment]
Thanks for the bug report, the help debugging, testing the various solutions, and verifying that this fix solved your issue! I have included this in 2.0.3, available now.
